keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -172,27 +172,50 @@ type PersistentVolumeClaimSpec struct {
   ...
 }
 
+// Add the PersistentVolumeClaimModifyVolume condition to existing PersistentVolumeClaimConditionType
+// Condition is used to document the last error controller sees
+const (
+  ...
+  // PersistentVolumeClaimModifyVolume - a user trigger modify volume of pvc has been started
+  PersistentVolumeClaimModifyVolume PersistentVolumeClaimConditionType = "ModifyVolume"
```

Comment:
I'm having trouble understanding this name of the new condition type. Is this saying that all the moditication of the volume which includes the reszie operation or just for changing volume attributes? If it's the latter, I think we should rename it to something like `ModifyVolumeAttributes`.

Reply:
I do not have a strong opinion of the naming, I was following the previous naming by using actions instead of a noun. This condition is used to document the last error the controller sees.

Reply:
Discussed in the sig-storage meeting, we will name it "VolumeAttriburesModifyError"

Reply:
If we are not going to change status when we retry `ModifyVolume` after failing (and status is set to `Infeasible`), then we need to somehow communicate to the user that, we are again retrying `ModifyVolume` (albeit slowly), so we will I think need a condition for that too. cc @msau42

Reply:
When do you propose we set a ModifyVolume condition to true and false? From [api-conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties), it says:
```
For state transitions which take a long period of time (e.g. more than 1 minute), it is reasonable to treat the transition itself as an observed state. In these cases, the Condition (such as "Resizing") itself should not be transient, and should instead be signalled using the True/False/Unknown pattern. This allows other observers to determine the last update from the controller, whether successful or failed. In cases where the state transition is unable to complete and continued reconciliation is not feasible, the Reason and Message should be used to indicate that the transition failed.
```

I interpret the last sentence to mean that if we still intend to retry, the status should still be True.

So the main benefits I see of having the condition is:
1) that the timestamp can be updated every time we retry an operation.
2) we could potentially return false for some preconditions like VAC doesn't exist or is invalid, PVC not bound, etc.